server/cloudapps/form.go

```diff
@@ -0,0 +1,71 @@
+package cloudapps
+
+type Form struct {
+	// RefreshOnChangeTo indicates that changes to the listed fields must reload
+	// the form. Values of the fields with values that are not included in the
+	// refreshed form are lost. Values that no longer apply are reset.
+	RefreshOnChangeTo []string `json:"refresh_on_change_to,omitempty"`
+	RefreshURL        string
+
+	Elements []interface{} // of *XXXElement
+}
+
+type ElementType string
+
+const (
+	ElementTypeCommand       = ElementType("command")
+	ElementTypeText          = ElementType("text")
+	ElementTypeStaticSelect  = ElementType("static_select")
+	ElementTypeDynamicSelect = ElementType("dynamic_select")
+	// ElementTypeTime          = ElementType("time")
+	ElementTypeBool    = ElementType("bool")
+	ElementTypeUser    = ElementType("user")
+	ElementTypeChannel = ElementType("channel")
+)
+
+type elementProps struct {
+	Type ElementType `json:"type"`
+
+	// Name is the name of the JSON field to use when submitting
+	Name string `json:"name"`
+
+	// Description is long description, can be used in modals in addition to
+	// Label, and in Autocomplete as Help
+	Description string `json:"description,omitempty"`
+
+	IsRequired bool `json:"is_required,omitempty"`
```

Comment:
What exactly would the semantics be? Also, should it be `IsNotClearable` (better defaulting? depends on the semantics. Is this a React thing, or common-sense UX (asking in case there are naming implications)

Reply:
None of the select boxes in the interactive dialog are clearable at the moment. 0/5 on how the developer can convey they want it clearable or not, but it needs to be supported. Maybe we should have them be clearable and not have the config field for it.